lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1143,7 +1143,7 @@ public function walkJoin($join)
                 $dqlAlias   = $joinDeclaration->aliasIdentificationVariable;
                 $tableAlias = $this->getSQLTableAlias($class->table['name'], $dqlAlias);
                 $condition = '(' . $this->walkConditionalExpression($join->conditionalExpression) . ')';
-                $isUnconditionalJoin = empty($conditions);
+                $isUnconditionalJoin = empty($condition);
```

Comment:
Wouldn't this be always true given the previous line always add brackets to the `$condition` variable meaning it will never be empty ?